File: src/main/java/alfheim/common/item/equipment/armor/elemental/ElementalArmor.java
package alfheim.common.item.equipment.armor.elemental;

import alfheim.api.*;
import alfheim.client.model.armor.ModelElementalArmor;
import alfheim.common.core.registry.AlfheimItems;
import cpw.mods.fml.relauncher.*;
import net.minecraft.client.model.ModelBiped;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import net.minecraft.util.StatCollector;
import vazkii.botania.api.item.IPixieSpawner;
import vazkii.botania.common.core.handler.ConfigHandler;
import vazkii.botania.common.item.ModItems;
import vazkii.botania.common.item.equipment.armor.manasteel.ItemManasteelArmor;

import java.util.List;

public abstract class ElementalArmor extends ItemManasteelArmor implements IPixieSpawner {

	public ElementalArmor(int type, String name) {
		super(type, name, AlfheimAPI.ELEMENTAL);
	}
	
	@Override
	@SideOnly(Side.CLIENT)
	public ModelBiped provideArmorModelForSlot(ItemStack stack, int slot) {
		models[slot] = new ModelElementalArmor(slot);
		return models[slot];
	}
	
	@Override
	public String getArmorTextureAfterInk(ItemStack stack, int slot) {
		return ModInfo.MODID + ":textures/model/armor/ElementalArmor_" + (ConfigHandler.enableArmorModels ? "new" : armorType == 2 ? "1" : "0") + ".png";
	}

	@Override
	public boolean getIsRepairable(ItemStack par1ItemStack, ItemStack par2ItemStack) {
		return par2ItemStack.getItem() == ModItems.manaResource && par2ItemStack.getItemDamage() == 7 || super.getIsRepairable(par1ItemStack, par2ItemStack);
	}

	static ItemStack[] armorset;

	@Override
	public ItemStack[] getArmorSetStacks() {
		if(armorset == null)
			armorset = new ItemStack[] {
				new ItemStack(AlfheimItems.elementalHelmet),
				new ItemStack(AlfheimItems.elementalChestplate),
				new ItemStack(AlfheimItems.elementalLeggings),
				new ItemStack(AlfheimItems.elementalBoots)
		};

		return armorset;
	}

	@Override
	public boolean hasArmorSetItem(EntityPlayer player, int i) {
		ItemStack stack = player.inventory.armorInventory[3 - i];
		if(stack == null)
			return false;

		switch(i) {
			case 0: return stack.getItem() == AlfheimItems.elementalHelmet || stack.getItem() == AlfheimItems.elementalHelmetRevealing;
			case 1: return stack.getItem() == AlfheimItems.elementalChestplate;
			case 2: return stack.getItem() == AlfheimItems.elementalLeggings;
			case 3: return stack.getItem() == AlfheimItems.elementalBoots;
		}

		return false;
	}

	@Override
	public String getArmorSetName() {
		return StatCollector.translateToLocal("alfheim.armorset.elemental.name");
	}

	@Override
	public void addArmorSetDescription(ItemStack stack, List<String> list) {
		super.addArmorSetDescription(stack, list);
		addStringToTooltip(StatCollector.translateToLocal("botania.armorset.elementium.desc"), list);
	}
}

File: src/main/java/alfheim/common/item/equipment/armor/elemental/ElementalArmor.kt
package alfheim.common.item.equipment.armor.elemental

import alexsocol.asjlib.meta
import alfheim.api.*
import alfheim.client.model.armor.ModelElementalArmor
import alfheim.common.core.util.AlfheimTab
import alfheim.common.item.AlfheimItems
import cpw.mods.fml.relauncher.*
import net.minecraft.client.model.ModelBiped
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.ItemStack
import net.minecraft.util.StatCollector
import vazkii.botania.api.item.IPixieSpawner
import vazkii.botania.common.core.handler.ConfigHandler
import vazkii.botania.common.item.ModItems
import vazkii.botania.common.item.equipment.armor.manasteel.ItemManasteelArmor

abstract class ElementalArmor(type: Int, name: String): ItemManasteelArmor(type, name, AlfheimAPI.elementalArmor), IPixieSpawner {
	
	init {
		creativeTab = AlfheimTab
	}
	
	@SideOnly(Side.CLIENT)
	override fun provideArmorModelForSlot(stack: ItemStack?, slot: Int): ModelBiped {
		models[slot] = ModelElementalArmor(slot)
		return models[slot]
	}
	
	override fun getArmorTextureAfterInk(stack: ItemStack?, slot: Int) =
		"${ModInfo.MODID}:textures/model/armor/ElementalArmor_${(if (ConfigHandler.enableArmorModels) "new" else if (armorType == 2) "1" else "0")}.png"
	
	override fun getIsRepairable(stack: ItemStack?, material: ItemStack) =
		material.item === ModItems.manaResource && material.meta == 7
	
	override fun getArmorSetStacks(): Array<ItemStack> {
		if (armorset == null)
			armorset = arrayOf(ItemStack(AlfheimItems.elementalHelmet), ItemStack(AlfheimItems.elementalChestplate), ItemStack(AlfheimItems.elementalLeggings), ItemStack(AlfheimItems.elementalBoots))
		
		return armorset!!
	}
	
	override fun hasArmorSetItem(player: EntityPlayer, i: Int): Boolean {
		val stack = player.inventory.armorInventory[3 - i] ?: return false
		
		when (i) {
			0 -> return stack.item === AlfheimItems.elementalHelmet || AlfheimItems.elementalHelmetRevealing?.let { stack.item === it } ?: false
			1 -> return stack.item === AlfheimItems.elementalChestplate
			2 -> return stack.item === AlfheimItems.elementalLeggings
			3 -> return stack.item === AlfheimItems.elementalBoots
		}
		
		return false
	}
	
	override fun getArmorSetName(): String {
		return StatCollector.translateToLocal("alfheim.armorset.elemental.name")
	}
	
	override fun addArmorSetDescription(stack: ItemStack?, list: List<String>) {
		super.addArmorSetDescription(stack, list)
		addStringToTooltip(StatCollector.translateToLocal("botania.armorset.elementium.desc"), list)
	}
	
	companion object {
		
		internal var armorset: Array<ItemStack>? = null
	}
}